Lykke.Bil2.Bcn.SignService/src/Lykke.Bil2.Bcn.SignService/Services/AddressGenerator.cs

```diff
@@ -0,0 +1,49 @@
+using System;
+using System.Threading.Tasks;
+using Lykke.Bil2.Bcn.SignService.Settings;
+using Lykke.Bil2.Contract.Common;
+using Lykke.Bil2.Contract.SignService.Requests;
+using Lykke.Bil2.Contract.SignService.Responses;
+using Lykke.Bil2.Sdk;
+using Lykke.Bil2.Sdk.SignService.Models;
+using Lykke.Bil2.Sdk.SignService.Services;
+
+namespace Lykke.Bil2.Bcn.SignService.Services
+{
+    public class AddressGenerator : IAddressGenerator
+    {
+        public AddressGenerator(/* TODO: Provide specific settings and dependecies, if necessary */)
+        {
+        }
+
+        public async Task<AddressCreationResult> CreateAddressAsync()
+        {
+            // Create dedicated deposit wallet,
+            // or return shared deposit wallet address.
+            //
+            // For example when shared deposit wallet scheme is used:
+            //
+            // return new AddressCreationResult(_hotWalletAddress, null, _hotWalletPublicKey.ToBase58());
```

Comment:
Mention about OperationNotSupportedException in case if offlice address creation is not supported by the blockchain.

Reply:
Fixed